src/changelogs/fetch_changelogs.py

```diff
@@ -5,53 +5,77 @@
 from bs4 import BeautifulSoup
 from urllib import request
 from utils import json_utils
+from typing import TypedDict
+
+
+class Changelog(TypedDict):
+    date: str
+    link: str
+
+
+class ChangelogLine(TypedDict):
+    Description: str
+    Tags: list[str]
 
 
 class ChangelogFetcher:
-    def __init__(self):
-        self.changelogs_by_version = {}
-        # layer1: version # as assigned by developers
-        # layer2: content (reworked in another pr, not gonna bother writing temporary
-        # stuff here for layer2-4)
+    """
+    Fetches changelogs from the deadlock forums and parses them into a dictionary
+    """
 
-        self.changelog_date_links = {}
-        # layer1: version # as assigned by developers
-        # layer2: date
+    # Hero lab changelogs need to be added manually to
+    # ./input-data/raw-changelogs following the
+    # naming convention 'herolab_2024_10_29.txt'
+    # see the referenced file for example formatting
+
+    # Then they need to be added to
+    # /output-data/changelogs/changelogs.json following
+    # the same naming convention and formatted the same
+    # as other changelogs, but with "forum_id": null
```

Comment:
I'm not fond of TODOs in the code. Just put all this info in a git issue

Reply:
This is instructions for adding each hero lab changelog whenever they are released, so not a TODO perse. See my question in the last sentence of my summary on the 29th. Still hopeful for better ideas

Reply:
Fair the first paragraph seems helpful. Maybe that could go into a README for `input-data`? As that's where someone will be when adding the hero labs changelog

On the second paragraph, it sounds like you're informing the dev to add something to output-data? Nothing should be manually modified in output-data, as it will just be overwritten by the parser

Reply:
The answer was so obvious, I should've realized thats why it felt so unintuitive lol. Moved that to input-data and added readme for input-data that has for now the changelog information.